edx_exams/apps/api/v1/views.py

```diff
@@ -1 +1,171 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+import logging
+import uuid
+
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from edx_api_doc_tools import (
+    path_parameter,
+    schema,
+    query_parameter,
+)
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+
+from edx_exams.apps.api.permissions import StaffUserPermissions
+from edx_exams.apps.api.serializers import ExamSerializer
+from edx_exams.apps.core.models import CourseExamConfiguration, Exam
+
+
+log = logging.getLogger(__name__)
+
+
+class ExamsView(APIView):
+    """
+    View to modify exams.
+
+    HTTP PATCH
+    Creates a new Exam.
+    Expected PATCH data: {
+        "course_id": "courses-v1:edx+test+2019",
+        "content_id": 123,
+        "exam_name": "Test Examination",
+        "time_limit_mins": 30,
+        "due_date": "2021-08-04",
+        "exam_type": "proctored",
+        "hide_after_due": False,
+        "is_active": True
+    }
+    **PATCH data Parameters**
+        * course_id: The unique identifier for the course.
+        * content_id: This will be the pointer to the id of the piece of course_ware which is the proctored exam.
+        * exam_name: This is the display name of the Exam (Midterm etc).
+        * time_limit_mins: Time limit (in minutes) that a student can finish this exam.
+        * due_date: The date on which the exam is due
+        * exam_type: Type of exam, i.e. timed, proctored
+        * hide_after_due: Whether the exam will be hidden from the learner after the due date has passed.
+        * is_active: Whether this exam will be active.
+    **Exceptions**
+        * HTTP_400_BAD_REQUEST
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (StaffUserPermissions,)
+
+    @schema(
+        body=ExamSerializer(many=True),
+        parameters=[
+            path_parameter('course_id', str, 'edX course run ID or external course key'),
+        ],
+        responses={
+            200: "OK",
+            400: "Invalid request. See message."
+        },
+        summary='Modify exams',
+        description='This endpoint should create new exams, update existing exams, '
+                    'and mark any active exams not included in the payload as inactive.'
+    )
+    def patch(self, request, course_id):
+        """ create or update an exam """
+        request_exams = request.data
+
+        serializer = ExamSerializer(data=request_exams, many=True)
+
+        if serializer.is_valid():
+            course_exams = Exam.objects.filter(course_id=course_id)
+
+            visited_exams = []
```

Comment:
maybe active_exams or still_relevant_exams or dont_mark_inactive

actually it's not even exams in here
content_with_exam maybe